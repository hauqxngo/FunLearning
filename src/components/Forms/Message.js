import React from "react";

/** Shows messaging alerts after handling forms.
 *
 * { LoginForm, SignupForm, ProfileForm } -> Message
 */

const Message = ({ type = "danger", messages = [] }) => {
  return (
    <div className={`alert alert-${type} mt-3`} role="alert">
      {messages.map((err) => (
        <p className="mb-0 small" key={err}>
          {err}
        </p>
      ))}
    </div>
  );
};

export default Message;
