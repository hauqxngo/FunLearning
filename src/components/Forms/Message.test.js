import React from "react";
import { render } from "@testing-library/react";
import Message from "./Message";

it("renders without crashing", () => {
  render(<Message />);
});

it("matches snapshot for danger", () => {
  let messages = ["You're about to delete it.", "Redrum."];
  const { asFragment } = render(<Message type="danger" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot for success", function () {
  let messages = ["Good to go!"];
  const { asFragment } = render(<Message type="success" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});
