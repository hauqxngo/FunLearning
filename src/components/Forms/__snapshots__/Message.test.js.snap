// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for danger 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger mt-3"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      You're about to delete it.
    </p>
    <p
      class="mb-0 small"
    >
      Redrum.
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success mt-3"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      Good to go!
    </p>
  </div>
</DocumentFragment>
`;
